packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js

```diff
@@ -0,0 +1,128 @@
+        'Elements with an interactive role and interaction handlers (mouse or key press) must be focusable.',
+      category: 'Fill me in',
+      recommended: false,
+    },
+    messages: {
+      elementWithRoleMustBe: "Elements with the '{{role}}' interactive role must be {{type}}.",
+    },
+    fixable: null, // or "code" or "whitespace"
+    schema: [
+      // fill in your schema
+    ],
+  },
+
+  create(context) {
+    // variables should be defined here
+
+    //----------------------------------------------------------------------
+    // Helpers
+    //----------------------------------------------------------------------
+
+    const tabbable = (context.options && context.options[0] && context.options[0].tabbable) || [];
+    const interactiveProps = [...eventHandlersByType.mouse, ...eventHandlersByType.keyboard];
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      TaggedTemplateExpression(node) {
+        // 1. only target html`` tagged template literals
+        if (isHtmlTaggedTemplate(node)) {
+          const analyzer = TemplateAnalyzer.create(node);
+
+          analyzer.traverse({
+            enterElement(element) {
+              // TODO: review this statement
+              // if (!includes(domElements, type)) {
+              //   // Do not test higher level JSX components, as we do not know what
+              //   // low-level DOM element this maps to.
+              //   return;
+              // }
+
+              const { attribs, name } = element;
+
+              const hasInteractiveProps = Object.keys(attribs).some(attr =>
+                interactiveProps.includes(attr),
+              );
+
+              if (
+                !hasInteractiveProps ||
+                isDisabledElement(attribs) ||
+                isHiddenFromScreenReader(name, attribs) ||
+                isPresentationRole(attribs)
+              ) {
+                // Presentation is an intentional signal from the author that this
+                // element is not meant to be perceivable. For example, a click screen
+                // to close a dialog.
+                return;
+              }
+
+              const hasTabindex = getTabIndex(getAttrVal(attribs.tabindex)) !== undefined;
+
+              if (
+                hasInteractiveProps &&
+                isInteractiveRole(attribs) &&
+                !isInteractiveElement(name, attribs) &&
+                !isNonInteractiveElement(name, attribs) &&
+                !isNonInteractiveRole(name, attribs) &&
+                !hasTabindex
+              ) {
+                const role = getAttrVal(attribs.role);
+
+                if (tabbable.includes(role)) {
+                  // Always tabbable, tabIndex = 0
+                  context.report({
+                    node,
+                    messageId: 'elementWithRoleMustBe',
```

Comment:
How will this report in the IDE? what does the error message look like? 🤔

Reply:
see line 33
```js
    messages: {
      elementWithRoleMustBe: "Elements with the '{{role}}' interactive role must be {{type}}.",
    },
```